tracer/src/Datadog.Trace.Coverage.collector/AssemblyProcessor.cs

```diff
@@ -694,25 +694,27 @@ public CustomResolver(ICollectorLogger logger, string assemblyFilePath)
                 _defaultResolver = new DefaultAssemblyResolver();
             }
 
-            public override AssemblyDefinition Resolve(AssemblyNameReference name)
+            public override AssemblyDefinition? Resolve(AssemblyNameReference name)
             {
-                AssemblyDefinition assembly;
+                AssemblyDefinition? assembly = null;
                 try
                 {
                     assembly = _defaultResolver.Resolve(name);
                 }
-                catch (AssemblyResolutionException ex)
+                catch (AssemblyResolutionException arEx)
                 {
                     var tracerAssemblyName = TracerAssembly.GetName();
                     if (name.Name == tracerAssemblyName.Name && name.Version == tracerAssemblyName.Version)
                     {
-                        if (!string.IsNullOrEmpty(_tracerAssemblyLocation))
+                        var cAssemblyLocation = !string.IsNullOrEmpty(_tracerAssemblyLocation) ? _tracerAssemblyLocation : TracerAssembly.Location;
+                        try
                         {
-                            assembly = AssemblyDefinition.ReadAssembly(_tracerAssemblyLocation);
+                            assembly = AssemblyDefinition.ReadAssembly(cAssemblyLocation);
                         }
-                        else
+                        catch (Exception innerAssemblyException)
                         {
-                            assembly = AssemblyDefinition.ReadAssembly(TracerAssembly.Location);
+                            _logger.Error(innerAssemblyException, $"Error reading the tracer assembly: {cAssemblyLocation}");
```

Comment:
nit: don't use interpolation in log messages
```suggestion
                            _logger.Error(innerAssemblyException, "Error reading the tracer assembly: {AssemblyLocation}", cAssemblyLocation);
```

Reply:
The logger used here is not the same:

https://github.com/DataDog/dd-trace-dotnet/blob/e064fdb419694a846f2a178658d4726c7b806c79/tracer/src/Datadog.Trace.Coverage.collector/ICollectorLogger.cs